modules/drivers/mongo/test/metabase/driver/mongo_test.clj

```diff
@@ -488,3 +489,19 @@
                (mt/rows
                 (mt/run-mbql-query bird_species
                   {:filter [:contains $bird_species._ "nett"]}))))))))
+
+(deftest strange-versionArray-test
+  (mt/test-driver :mongo
+    (testing "Negative values in versionArray are ignored (#29678)"
+      (with-redefs [mg/command (constantly {"version" "4.0.28-23"
+                                            "versionArray" [4 0 29 -100]})]
+        (is (= {:version "4.0.28-23"
+                :semantic-version [4 0 29]}
+               (driver/dbms-version :mongo (mt/db))))))
+
+    (testing "Any values after rubbish in versionArray are ignored"
+      (with-redefs [mg/command (constantly {"version" "4.0.28-23"
```

Comment:
i see where the negative numbers come from. I'm surprised the solution wasn't to parse this as `[4 0 28]` instead of sanitizing `[4 0 "NaN" 29]` or `[4 0 "NaN" -29]`.

For instance, my h2 version is
```clojure
{:flavor "H2",
 :version "2.1.214 (2022-06-13)",
 :semantic-version [2 1]}
```

and i'm surprised to see it as `[2 1]` and not `[2 1 214]`. Is that too complicated of a fix?

Reply:
This is because whenever a driver offers a machine processable (i.e., parsed) version, we use that as semantic version.  The JDBC driver and Mongo provide such versions.

I agree it's ugly that the human readably and the machine processable versions are out of sync, but the source of that is in the driver.